### PR TITLE
add a formats option to the cache task to spit version file in edn or clj/cljs/cljx format

### PR DESCRIPTION
**Pull request by jgrodziski**
Hi Chris,

Here is an adapted version taking into account your comments allowing better flexibility regarding the version file generation while maintaining backward compatibility with the previous behavior (only clj file generated by cache).

Jérémie.

### CODE REVIEW

**Comment by cch1** [comment]
I'm OK with a new PR (this one or yet another one).  But please resubmit with commit messages structured per this : https://chris.beams.io/posts/git-commit/

**Comment by jgrodziski** [comment]
Ok I'll do that, and I'll update the README at the same time.

**Comment by jgrodziski** [comment]
Hi Chris, here is the PR with an appropriate commit message and the updated README.

**Comment by cch1** [comment]
On #19, I'm still seeing the same commits with the same commit messages + one new merge commit.

**Comment by jgrodziski** [comment]
Ok, the commit is now unique and clean (fix up the mess thanks to rebase and forced push :)).

File: README.md [hunk 1]
```diff
@@ -22,9 +22,50 @@ There are two lein sub-tasks within the v namespace intended for direct use:
 Show the effective version of the project and workspace state.
 
 ### lein v cache
-Cache the effective version of the project to a file (default is `version.clj`) in the first source directory (typically `src`).  It is possible to have the version cached to a file automatically by defining a prep task in your project:
+Cache the effective version of the project to a file (default is `version.clj`) in the first source directory (typically `src`). It is possible to have the version cached to a file automatically by defining a prep task in your project like this:
 
     :prep-tasks [["v" "cache" "src"]]
+    
+The `lein v cache` command has a variadic argument that is the following: first is the directory to output to (default to `src`), then the rest is the list of the output sufix you wish and hence format (default to `clj`). The available suffixes are: clj, cljs, cljx, cljc and edn.
```
Comment (cch1) [inline README.md]: Typo: sufix => suffix

File: README.md [hunk 2]
```diff
@@ -22,9 +22,50 @@ There are two lein sub-tasks within the v namespace intended for direct use:
...
+```sh
+lein v cache src/cljs cljs
+# => output only the src/cljs/version.cljs file
+lein v cache src/clj clj edn
+# => output the src/clj/version.clj and src/clj/version.edn file
+lein v cache src edn
+# => output the src/version.edn file
+```
+
+#### Example of Clojure source code version output
```
Comment (cch1) [inline README.md]: For all of these examples, it would be better to not use a `-DIRTY` version.

Comment (jgrodziski) [reply README.md]: ok, the versions used in examples have the -DIRTY removed.

**Pull request merged**